src/natcap/invest/managed_raster/managed_raster.pxd

```diff
@@ -37,3 +46,16 @@ cdef class _ManagedRaster:
     cdef inline void set(_ManagedRaster self, long xi, long yi, double value)
     cdef inline double get(_ManagedRaster self, long xi, long yi)
     cdef void _load_block(_ManagedRaster self, int block_index) except *
+
+
+cdef class ManagedFlowDirRaster(_ManagedRaster):
+
+    cdef bint is_local_high_point(ManagedFlowDirRaster self, long xi, long yi)
+
+    cdef vector[NeighborTuple] get_upslope_neighbors(ManagedFlowDirRaster self, long xi, long yi)
+
+    cdef vector[NeighborTuple] get_downslope_neighbors(ManagedFlowDirRaster self, long xi, long yi, bint skip_oob=*)
+
+
+cdef inline int is_close(double x, double y):
+    return abs(x-y) <= (1e-8+1e-05*abs(y))
```

Comment:
Oh!  I just remembered a recent case in `pygeoprocessing` where we added in a check for whether `x` and/or `y` were `nan`s and returning an appropriate check accordingly.  Maybe we could save ourselves some future headaches by including such a check here?

Reply:
That would be conveninent, but do you think the additional overhead would be a problem? I assume that calling `math.isnan` or `numpy.isnan` would cause a slowdown

Reply:
You're right, they would!  But there should also be a `cimport`able `math` module with an `isnan` function, which should make the addition negligibly slower.  Looks like we added this over in pygeoprocessing's routing, so I bet we might be able to just use the same solution here: https://github.com/natcap/pygeoprocessing/blob/254c7c8fbbf5220e3d28d69e8cdf945fc8a6c3a9/src/pygeoprocessing/routing/routing.pyx#L197

Reply:
Oh neat! I added that check to the `is_close` in `managed_raster.pxd` for now, and eventually when we consolidate things into pygeoprocessing, those functions will probably be combined.

Reply:
Awesome, thanks!